alacritty_terminal/src/tty/windows/conpty.rs

```diff
@@ -230,6 +241,54 @@ pub fn new(config: &Options, window_size: WindowSize) -> Option<Pty> {
     Some(Pty::new(conpty, conout, conin, child_watcher))
 }
 
+// https://learn.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-createprocessa#parameters
+// https://learn.microsoft.com/sk-SK/previous-versions/troubleshoot/windows/win32/createprocess-cannot-eliminate-duplicate-variables#environment-variables
+fn convert_custom_env(custom_env: &HashMap<String, String>) -> Option<Vec<u16>> {
+    // Windows inherits parent's env when no `lpEnvironment` parameter is specified.
+    if custom_env.is_empty() {
+        return None;
+    }
+
+    let mut converted_block = Vec::new();
+    let mut all_env_keys = HashSet::new();
+    for (custom_key, custom_value) in custom_env {
+        let custom_key = OsStr::new(custom_key);
+        if all_env_keys.insert(custom_key.to_ascii_uppercase()) {
+            add_windows_env_key_value_to_block(
+                &mut converted_block,
+                custom_key,
+                OsStr::new(&custom_value),
+            );
+        }
+    }
+
+    // Pull the current a process environment after to not overwrite user provided one.
+    for (inherited_key, inherited_value) in std::env::vars_os() {
+        if all_env_keys.insert(inherited_key.to_ascii_uppercase()) {
+            add_windows_env_key_value_to_block(
+                &mut converted_block,
+                &inherited_key,
+                &inherited_value,
+            );
+        }
+    }
+
+    converted_block.push(0);
+    Some(converted_block)
+}
+
+// According to the `lpEnvironment` parameter description in the link above:
```

Comment:
Could you just move the link here instead?

Reply:
Sure, also added a few words about case-insensitivity for the remaining link.